dbeam-core/src/main/java/com/spotify/dbeam/options/KmsSecrets.java

```
@@ -0,0 +1,185 @@
+
+import com.google.api.client.googleapis.auth.oauth2.GoogleCredential;
+import com.google.api.client.googleapis.util.Utils;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.services.cloudkms.v1.CloudKMS;
+import com.google.api.services.cloudkms.v1.CloudKMSScopes;
+import com.google.api.services.cloudkms.v1.model.DecryptRequest;
+import com.google.api.services.cloudkms.v1.model.DecryptResponse;
+import com.google.auto.value.AutoValue;
+import com.google.cloud.ServiceOptions;
+import com.google.common.base.CharMatcher;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.Optional;
+import java.util.Properties;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class KmsSecrets {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(PasswordReader.class);
+  private static final String DEFAULT_KEYRING;
+  private static final String DEFAULT_KEY;
+  private static final String DEFAULT_LOCATION;
+
+  static {
+    Properties p = System.getProperties();
+    DEFAULT_KEYRING = p.getProperty("KMS_DEFAULT_KEYRING", "dbeam");
+    DEFAULT_KEY = p.getProperty("KMS_DEFAULT_KEY", "default");
+    DEFAULT_LOCATION = p.getProperty("KMS_DEFAULT_LOCATION", "global");
+  }
+
+  /**
+   * Create a new {@link Decrypter.Builder} for customizing the {@link Decrypter} configuration.
+   */
+  public static Decrypter.Builder decrypter() {
+    Decrypter.Builder decrypter = new AutoValue_KmsSecrets_Decrypter.Builder()
+        .location(DEFAULT_LOCATION)
+        .key(DEFAULT_KEY)
+        .keyring(DEFAULT_KEYRING)
+        .project(ServiceOptions.getDefaultProjectId());
+    try {
+      return decrypter.credentials(GoogleCredential.getApplicationDefault());
+    } catch (IOException e) {
+      LOGGER.warn("Unable to fetch default Google application credential");
+      return decrypter;
```

It might be better to fail loudly instead of returning an unusable decrypter.